### PR TITLE
`Development`: Improve and simplify local playwright setup

### PR DESCRIPTION
**Pull request by SimonEntholzer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Setting up playwright initially is quite cumbersome, as several steps need to be followed before it can be used.

### Description

This PR simplifies the setup by providing three simple scripts, the user can execute to run playwright fast.

### Steps for Testing
1. Locally, checkout this branch.  (`checkout feature/playwright/simplify-setup`)
2. See the README.md in `supporting_scripts/playwright`
3. Follow the steps, and verify that the steps are easy to follow, and you can run playwright tests in the UI.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1 on linux (ubuntu/debian, other distros might not work)
- [x] Test 2 on macOS
- [x] Test 3 on macOS

## Release Notes

- **New Features**
	- Enhanced documentation for Playwright setup, including a streamlined three-step process.
	- New scripts for automating user setup and running Artemis in Docker on both macOS and Linux.
	- Introduced a new script for installing and launching Playwright.

- **Improvements**
	- Simplified username and password template configuration.
	- Added timeout configuration options for better test execution control.
	- Updated installation command for Playwright to reflect changes in the setup process. 

- **Documentation**
	- Added a README file for clearer instructions on setting up Playwright with Artemis.

### CODE REVIEW

**Review by az108 — DISMISSED** [review]
Tested it locally, was able to install playwright and run the tests successfully 👍

**Review by coolchock — DISMISSED** [review]
tested on mac, works as expected

File: docs/dev/playwright.rst
Comment (BBesrour) [inline docs/dev/playwright.rst:1]: Could you also include your script in the documentation? 
For me, It’s important to clearly differentiate between the version that runs Artemis on Docker and the one that requires Artemis to be already running. 
Would be great to also outline the requirements and steps for both versions clearly.

**Review by BBesrour — CHANGES_REQUESTED** [review]

File: docs/dev/playwright.rst
Comment (SimonEntholzer) [reply docs/dev/playwright.rst:1]: I adapted the documentation, is it more clear now?

**Review by BBesrour — DISMISSED** [review]
code lgtm

**Review by muradium — CHANGES_REQUESTED** [review]
`docker/playwright.yml` compose file needs to be changed to correctly execute the tests on CI. Command of `artemis-playwright` should have `npm run playwright:install` now  instead of `npm run playwright:setup`. Also we probably need to execute `setupUsers.sh` script between `playwright:install` and `playwright:test` steps in compose file, now that setting up users is a separate script.

**Comment by SimonEntholzer** [comment]
> `docker/playwright.yml` compose file needs to be changed to correctly execute the tests on CI. Command of `artemis-playwright` should have `npm run playwright:install` now instead of `npm run playwright:setup`. Also we probably need to execute `setupUsers.sh` script between `playwright:install` and `playwright:test` steps in compose file, now that setting up users is a separate script.

That's a good point. I reverted the changes, which the CLI tests need to run on bamboo. For the user, using the setup locally nothing has changed.
To have a clear separation between local/bamboo (especially because we want to move to github for the e2e tests soon) I decided to not modify the current playwright-docker setup

**Review by az108 — APPROVED** [review]
reapprove after changes

**Review by BBesrour — APPROVED** [review]

File: docs/dev/playwright.rst [hunk 1]
```diff
@@ -29,40 +41,49 @@ Playwright tests rely on the Playwright Node.js library, browser binaries, and s
...
-    Install Playwright browser binaries, set up the environment to ensure Playwright can locate these binaries, and
-    create test users (if creating users is enabled in the configuration) with the following command:
+    Install Playwright browser binaries, set up the environment to ensure Playwright can locate these binaries.
+    On some operating systems this might not work, and playwright needs to be manually installed via a package manager.
 
     .. code-block:: bash
 
-        npm run playwright:setup
+        npm run playwright:setup-local
+        npm run playwright:init
```
Comment (muradium) [inline docs/dev/playwright.rst:84]: Do we still need this step during a local run? The users are already created with a supporting script in the previous step. I thought we left `playwright:init` script just for CI?

**Review by muradium — APPROVED** [review]
Thanks for the changes. Supporting scripts also work as expected 👍

**Comment by muradium** [comment]
Left a small comment. Also I added this branch manually to bamboo, as it wasn't created for some reason 😅

File: docs/dev/playwright.rst (hunk 1 above)
Comment (SimonEntholzer) [reply docs/dev/playwright.rst:84]: If you run it without the user generation script, that would still be needed (at least if the setup command does not complete correctly). So, if you do not use the simplified setup, you should still use this command.

**Review by magaupp — CHANGES_REQUESTED** [review]
The scripts would be more comfortable to use if they work regardless of the current working directory.

The target of the `cd` commands can be made more obvious if they use paths relative to Artemis' root directory.

**Review by magaupp — APPROVED** [review]
Thanks for the changes. Code looks good :+1:

**Review by muradium — APPROVED** [review]
Reapprove after changes

**Review by BBesrour — APPROVED** [review]

**Pull request merged**